modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/api/construction/ConstructionMetadata.scala

```diff
@@ -0,0 +1,21 @@
+package org.tessellation.rosetta.domain.api.construction
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{TransactionFee, TransactionReference, TransactionSalt}
+
+import derevo.cats.show
+import derevo.circe.magnolia.customizableDecoder
+import derevo.derive
+
+case object ConstructionMetadata {
+
+  @derive(customizableDecoder, show)
+  case class TransactionMetadata(
+    source: Address,
+    lastReference: TransactionReference,
+    fee: TransactionFee,
```

Comment:
metadata property for `/construction/metadata` specifies the separate field for `suggested_fee`. IMO you should use this field to suggest a fee, not custom field through metadata.

Reply:
@marcinwadon Can you please clarify what you mean?
E.g. Are you suggesting a simple rename from `fee` to `suggestedFee`, or also changing the type to `SuggestedFee`, or `NonEmptyList[SuggestedFee]` ?

Reply:
https://www.rosetta-api.org/docs/models/ConstructionMetadataResponse.html

What I mean is that `/construction/metadata` specifies response as two fields - metadata and suggested fee. Instead of using the suggested fee field, you created a custom field inside metadata property. You can use the suggested fee as described by the rosetta specification

Reply:
@marcinwadon That's what I thought you meant so I went and changed it to do that. However, in hindsight wouldn't it make more since to do it the way it was originally written? If the intent of the _metadata_ endpoint is to provide something to forward to the next call then aren't we now requiring the caller to map the response of one endpoint into something else to pass to the next?
https://www.rosetta-api.org/docs/ConstructionApi.html#constructionpayloads

Reply:
@marcinwadon My understanding is that currently we're a fee-less system so when the fee is missing it's ok to set it to zero -- I don't think it was zero all the time.

My concern was about the refactor to move the fee out of the `metadata` response.

Reply:
I just said that:
- rosetta specifies a field to suggest a fee
- fee for transaction should be selected by the client, not automatically by the node